Libraries/LibGUI/GTextEditor.cpp

```diff
@@ -932,18 +947,22 @@ void GTextEditor::update_cursor()
 
 void GTextEditor::update_undo_timer()
 {
-    if (m_undo_stack.size() <= 0)
+
+    if (m_undo_stack_size <= 0)
         return;
 
+    auto& undo_vector = m_undo_stack[0].m_undo_vector;
+
     if (m_undo_timer == 0)
-        m_prev_undo_stack_size = m_undo_stack[m_undo_index].size();
+        m_prev_undo_vector_size = undo_vector.size();
 
-    if (m_undo_timer >= 2 && m_undo_stack[m_undo_index].size() > 0) {
+    if (m_undo_timer >= 2 && undo_vector.size() > 0) {
 
-        if (m_undo_stack[m_undo_index].size() == m_prev_undo_stack_size) {
-            dbg() << "Increased Undo Index";
-            m_undo_stack.append(make<NonnullOwnPtrVector<UndoCommand>>());
-            m_undo_index++;
+        if (undo_vector.size() == m_prev_undo_vector_size) {
+            auto undo_commands_container = make<UndoCommandsContainer>();
+            undo_commands_container->m_undo_vector = {};
```

Comment:
You don't need to initialize this, it's already default-constructed on the line above it :)